### ISSUE TITLE
add typings for this lib

### ISSUE THREAD

**Issue by Diluka** [recombee/node-api-client#21]
add typescript typings for static type check and code autocomplete

**Comment by jbtheard** [comment]
Would be great to have this feature. We had to "decorate" every call with our own types...

**Comment by marhaupe** [comment]
This shouldn't even be too hard to do. There's even native tooling for it out there:
https://www.typescriptlang.org/docs/handbook/declaration-files/dts-from-js.html
I'm currently only trying out recombee, but if I move forward and use it I'll at least try to publish a custom `@types/recombee-api-client`, since not even that exists.

**Comment by vallpaper** [comment]
Hi! Thanks for bringing this up. We decided to prioritize this and release typed packages for both JS SDKs in the following weeks.

**Comment by Esarhaddon** [comment]
Glad to hear that types are coming soon. Here's a WIP `d.ts` file in the meantime which might be helpful for other Typescript users just trying to get up and running while we wait for the official types:

```
declare module "recombee-api-client" {
  const ApiClient: {
    new (
      databaseId: string,
      privateToken: string,
      options: { region: string },
    ): {
      send: <T = unknown>(
        request: unknown,
        callback?: (err: unknown, res: T) => void,
      ) => Promise<T>
    }
  }

  const requests: {
    Batch: {
      new (requests: unknown[]): unknown
    }
    ListItems: {
      new (options?: {
        offset?: number
        count?: number
        filter?: string
        returnProperties?: boolean
      }): unknown
    }
    SetItemValues: {
      new (
        itemId: string,
        values: Record<string, unknown>,
        options?: { cascadeCreate?: boolean },
      ): unknown
    }
    DeleteItem: {
      new (itemId: string): unknown
    }
    ListItemProperties: {
      new (): unknown
    }
    AddItemProperty: {
      new (name: string, type: string): unknown
    }
    DeleteItemProperty: {
      new (name: string): unknown
    }
    RecommendItemsToUser: {
      new (
        userId: string,
        count: number,
        options?: {
          scenario?: string
          cascadeCreate?: boolean
          returnProperties?: boolean
          filter?: string
        },
      )
    }
    RecommendItemsToItem: {
      new (
        itemId: string,
        targetUserId: string | null,
        count: number,
        options?: {
          scenario?: string
          cascadeCreate?: boolean
          returnProperties?: boolean
          filter?: string
        },
      )
    }
  }
}
```

**Comment by vallpaper** [comment]
Hi everyone, we released SDK version 4.1.2 with full TS support. If you find any issue, please report it :)
